IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.m

```diff
@@ -44,25 +44,27 @@ - (void)handleError:(NSError *)error
     BOOL shouldRetry = YES;
     shouldRetry &= httpRequest.retryCounter > 0;
     BOOL shouldRetryNetworkingFailure = NO;
-    if (!httpResponse && error)
+    if (!httpResponse)
     {
-        // Networking errors (-1003. -1004. -1005. -1009)
-        shouldRetryNetworkingFailure = shouldRetry && (error.code == NSURLErrorCannotFindHost || error.code == NSURLErrorCannotConnectToHost || error.code == NSURLErrorNetworkConnectionLost  || error.code == NSURLErrorNotConnectedToInternet);
-        if (!shouldRetryNetworkingFailure)
+        if (shouldRetry && error)
+        {
+            // Networking errors (-1003. -1004. -1005. -1009)
+            shouldRetryNetworkingFailure = error.code == NSURLErrorCannotFindHost || error.code == NSURLErrorCannotConnectToHost || error.code == NSURLErrorNetworkConnectionLost || error.code == NSURLErrorNotConnectedToInternet;
+        }
+
+        shouldRetry &= shouldRetryNetworkingFailure;
+        if (!shouldRetry)
         {
             if (completionBlock) completionBlock(nil, error);
             return;
         }
     }
-    else if (!httpResponse)
+    else if (shouldRetry)
```

Comment:
nit: moving the `if (shouldRetry)` inside the block may make it a bit easier to undertand :)

Reply:
Updated